autokeras/hypermodel/hyper_block.py

```diff
@@ -41,51 +43,43 @@ def build(self, hp, inputs=None):
-            default='act-bn-dense')
+            ['dense-bn-act', 'dense-act'],
+            default='dense-bn-act')
+        dropout_rate = hp.Choice('dropout_rate', [0, 0.25, 0.5], default=0.5)
         for i in range(hp.Choice('num_layers', [1, 2, 3], default=2)):
+            units = hp.Choice(
+                'units_{i}'.format(i=i),
+                [16, 32, 64, 128, 256, 512, 1024],
+                default=32)
             if layer_stack == 'dense-bn-act':
-                output_node = tf.keras.layers.Dense(hp.Choice(
-                    'units_{i}'.format(i=i),
-                    [16, 32, 64, 128, 256, 512, 1024],
-                    default=32))(output_node)
+                output_node = tf.keras.layers.Dense(units)(output_node)
                 output_node = tf.keras.layers.BatchNormalization()(output_node)
-                output_node = tf.keras.layers.Activation(active_category)(
-                    output_node)
-                output_node = tf.keras.layers.Dropout(
-                    rate=hp.Choice('dropout_rate', [0, 0.25, 0.5], default=0.5))(
-                    output_node)
+                output_node = tf.keras.layers.ReLU()(output_node)
+                output_node = tf.keras.layers.Dropout(dropout_rate)(output_node)
             elif layer_stack == 'dense-act':
-                output_node = tf.keras.layers.Dense(
-                    hp.Choice('units_{i}'.format(i=i),
-                              [16, 32, 64, 128, 256, 512, 1024],
-                              default=32))(output_node)
-                output_node = tf.keras.layers.Activation(active_category)(
-                    output_node)
-                output_node = tf.keras.layers.Dropout(
-                    rate=hp.Choice('dropout_rate', [0, 0.25, 0.5], default=0.5))(
-                    output_node)
-            else:
-                output_node = tf.keras.layers.Activation(active_category)(
-                    output_node)
-                output_node = tf.keras.layers.BatchNormalization()(output_node)
-                output_node = tf.keras.layers.Dense(
-                    hp.Choice('units_{i}'.format(i=i),
-                              [16, 32, 64, 128, 256, 512, 1024],
-                              default=32))(output_node)
-                output_node = tf.keras.layers.Dropout(
-                    rate=hp.Choice('dropout_rate', [0, 0.25, 0.5], default=0.5))(
-                    output_node)
+                output_node = tf.keras.layers.Dense(units)(output_node)
+                output_node = tf.keras.layers.ReLU()(output_node)
+                output_node = tf.keras.layers.Dropout(dropout_rate)(output_node)
         return output_node
 
 
 class RNNBlock(HyperBlock):
+    """ An RNN HyperBlock.
+
+    Attributes:
+        bidirectional: Boolean. If not provided, it would be a tunable variable.
+        return_sequences: Boolean.  If not provided, it would be a tunable variable.
```

Comment:
If it is text classification, it can either return a sequence and reduce the dimension, or directly return the vector. I think they can be alternatives.